src/Components/Calculator.js

```diff
@@ -1,40 +1,75 @@
-    </div>
-    <div className="buttons-div">
-      <button type="button">AC</button>
-      <button type="button">+/-</button>
-      <button type="button">%</button>
-      <button type="button" className="operation">÷</button>
-    </div>
-    <div className="buttons-div">
-      <button type="button">7</button>
-      <button type="button">8</button>
-      <button type="button">9</button>
-      <button type="button" className="operation">x</button>
-    </div>
-    <div className="buttons-div">
-      <button type="button">4</button>
-      <button type="button">5</button>
-      <button type="button">6</button>
-      <button type="button" className="operation">-</button>
-    </div>
-    <div className="buttons-div">
-      <button type="button">1</button>
-      <button type="button">2</button>
-      <button type="button">3</button>
-      <button type="button" className="operation">+</button>
-    </div>
-    <div className="buttons-div">
-      <button type="button" className="zero">0</button>
-      <button type="button">.</button>
-      <button type="button" className="operation">=</button>
-    </div>
-  </div>
+const Button = ({ label, type, handleClick }) => (
+  <button type="button" className={type} onClick={handleClick}>
+    {label}
+  </button>
 );
 
+const Calculator = () => {
+  const [state, setState] = useState({
+    total: null,
+    next: null,
+    operation: null,
+  });
+
+  const handleClick = (buttonName) => {
+    const newState = calculate(state, buttonName);
+    setState(newState);
+  };
+
+  const renderButton = (label, type) => (
+    <Button key={label} label={label} type={type} handleClick={() => handleClick(label)} />
+  );
+
+  return (
+    <div className="calculator">
+      <div className="buttons-div">
+        <div className="input-field">{state.next || state.total || '0'}</div>
+      </div>
+      <div className="buttons-div">
+        {[
+          ['AC', 'operation'],
+          ['+/-', 'operation'],
+          ['%', 'operation'],
+          ['÷', 'operation'],
+        ].map(([label, type]) => renderButton(label, type))}
+      </div>
+      <div className="buttons-div">
+        {[
+          ['7', 'number'],
+          ['8', 'number'],
+          ['9', 'number'],
+          ['x', 'operation'],
+        ].map(([label, type]) => renderButton(label, type))}
+      </div>
+      <div className="buttons-div">
+        {[
+          ['4', 'number'],
+          ['5', 'number'],
+          ['6', 'number'],
+          ['-', 'operation'],
+        ].map(([label, type]) => renderButton(label, type))}
+      </div>
+      <div className="buttons-div">
+        {[
+          ['1', 'number'],
+          ['2', 'number'],
+          ['3', 'number'],
+          ['+', 'operation'],
+        ].map(([label, type]) => renderButton(label, type))}
+      </div>
+      <div className="buttons-div">
+        {[
+          ['0', 'number zero'],
+          ['.', 'operation'],
```

Comment:
- [ ] Kindly fix the `.` operation button as it's working. When I press the number then I press the `.` operation it displays like that:
![image](https://github.com/kkmanuu/math-magicians/assets/24703171/a9d2f2a4-842e-4fd3-9238-98785d9f5d16) and lead to an error as it's not a valid number.

Reply:
@gemmen29 I have fix the operation button and it's now working as expected.